chunky/src/java/se/llbit/chunky/renderer/scene/AsynchronousSceneManager.java

```diff
@@ -39,16 +40,39 @@
  *
  * @author Jesper Öqvist <jesper@llbit.se>
  */
-public class AsynchronousSceneManager extends Thread implements SceneManager {
+public class AsynchronousSceneManager implements SceneManager {
 
   private final SynchronousSceneManager sceneManager;
-  private final LinkedBlockingQueue<Runnable> taskQueue;
+  private final ExecutorService executorService = Executors.newSingleThreadExecutor(runnable -> {
+    Thread thread = new Thread(runnable, "AsynchronousSceneManager-Thread");
+    thread.setUncaughtExceptionHandler((t, exception) -> {
+      if (exception instanceof OutOfMemoryError) {
+        Log.error(
+          "Chunky has run out of memory! Increase the memory given to Chunky in the launcher.",
+          exception);
+        throw (Error) exception;
+      } else if(exception instanceof InterruptedException) {
+        Thread.currentThread().interrupt();
+      } else {
+        Log.error("Scene manager has crashed due to an uncaught exception. \n" +
+          "Chunky might not work properly until you restart it.\n" +
+          "If you think this is a bug, please report it and the following lines to the developers.",
+          exception);
+        // a new thread will be created by the executor after the current one terminated and a new task is enqueued
+      }
+    });
+    return thread;
+  });
 
   public AsynchronousSceneManager(RenderContext context, RenderManager renderManager) {
-    super("Scene Manager");
-
     sceneManager = new SynchronousSceneManager(context, renderManager);
-    taskQueue = new LinkedBlockingQueue<>();
+  }
+
+  /**
+   * Schedule a task to be run soon.
```

Comment:
Maybe we should add a note on why one would want to use this, i.e. to safely modify the scene without race conditions?

Reply:
Feel free to suggest a descriptive JavaDoc ~ I'm not sure myself when to use a task with this, I just updated the code ^^
Seems like all operations in the AsyncSceneManager are enqueued with this.

Reply:
Probably because that allows it to be async and not do expensive operations in the UI thread. :brain:

Reply:
Well yes, I see that 😅 
But this seems too obvious to state in a note here (?) Did you have anything else in mind? 🙈